### PR TITLE
Peer Code Review - Pull Request

### PR DESCRIPTION
**Pull request by gabrielwright1**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,199 @@
-import './App.css';
+// styling
+import "./styles/sass/App.scss";
+
+// modules
+import { useEffect, useState } from "react";
+import {
```
Comment (Cristina-Codes) [inline src/App.js:6]: Amazing to see you utilize so much of Firebase's offerings and lay out the imports clearly.

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,199 @@
...
+		const { name } = burgerProduct;
+		setModalBurger(name);
+	};
+
+	return (
+		<div className="App">
+			<Modal modalBurger={modalBurger} />
+			<div className="bg-img-container wrapper">
+				{/* Modal displays burger customization options */}
+				<header>
```
Comment (Cristina-Codes) [inline src/App.js]: You might consider creating a Header component or placing part of the header in a component to shrink this portion down a bit.

File: src/App.js [hunk 3]
```diff
@@ -1,11 +1,199 @@
...
+							<div className="img-container">
+								<img
+									src="https://images.unsplash.com/photo-1513918573039-35f42ae13895?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1074&q=80"
+									alt="an old 50s car in front of a diner"
+								/>
+							</div>
+							<ShoppingCart
+								burgers={burgers}
+								total={total}
+								handleRemove={handleRemove}
```
Comment (Cristina-Codes) [inline src/App.js]: You remembered not to use the anon function here since we are not passing in arguments - nicely done 🙌🏻

File: src/App.js [hunk 4]
```diff
@@ -1,11 +1,199 @@
...
+								handleCheckout={handleCheckout}
+							/>
+						</div>
+					</div>
+				</header>
+				<main>
```
Comment (Cristina-Codes) [inline src/App.js]: Similar idea here - could you create a component for the main or parts of it to reduce the size of this.

File: src/components/Card.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
+// modules
+import { useState, useEffect } from "react";
+
+const Card = ({ addListItems, burgerProduct, handleModalOpen }) => {
+	// state
+	const [counter, setCounter] = useState(0);
+	const [subtotal, setSubtotal] = useState(0);
+
+	// destructure burger
+	const { name, imgUrl, price } = burgerProduct;
```
Comment (Cristina-Codes) [inline src/components/Card.js:10]: You seem to have a great handle on destructuring

File: src/components/Card.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+		if (counter > 0) {
+			setCounter(counter - 1);
+		}
+		handleUnclick(e);
+	};
+	const handleSubmit = (e) => {
+		e.preventDefault();
+
+		// check to see if user selected counter
+		if (counter > 0) {
```
Comment (Cristina-Codes) [inline src/components/Card.js:32]: Gotta check for those user errors! Nicely done here with the addition of the alert to bring attention to the problem immediately. If they are verging on hangry, they need the quick solve.

File: src/components/Card.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+		<form
+			onSubmit={(e) => {
+				handleSubmit(e);
+			}}
+			action="submit"
+			className="burger-card"
+		>
+			<div className="img-container">
+				<label htmlFor="open-modal" className="sr-only">
+					Click to customize the burger
```
Comment (Cristina-Codes) [inline src/components/Card.js:60]: Clear message to support screen readers 🙌🏻

File: src/styles/sass/partials/_typography.scss [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+h1 {
+	@include setTypography($ff-primary, $fw-regular, $fs-800);
+	color: $clr-white;
+	background-color: $clr-black;
+	text-shadow: (4px 4px 4px $clr-dark-blue);
+	padding: 10px 0;
+	margin: 30px 0;
+}
+
+// h2 - my toolkit, recent projects, my journey
```
Comment (Cristina-Codes) [inline src/styles/sass/partials/_typography.scss]: Was this comment for your portfolio?

File: src/styles/sass/partials/_typography.scss [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+	color: $clr-white;
+	text-shadow: (1px 1px 4px $clr-pink);
+	padding: 10px;
+	text-align: center;
+	margin: 20px 0;
+	background-color: $clr-black;
+	width: 100%;
+}
+
+// h3 - I'm a front end, who am i here are some, Hacker News, please feel free
```
Comment (Cristina-Codes) [inline src/styles/sass/partials/_typography.scss]: Here's another comment from your previous project

File: src/styles/sass/partials/_typography.scss [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+.sm-button {
+	@include setTypography($ff-secondary, $fw-bold, $fs-400);
+	color: $clr-white;
+	background-color: $clr-dark-pink;
+	border-radius: $border-radius;
+	border: none;
+	padding: 1rem 0.8rem;
+	// margin: 0 5px;
```
Comment (Cristina-Codes) [inline src/styles/sass/partials/_typography.scss]: You might remove this commented margin

**Review by Cristina-Codes — APPROVED** [review]
Now I'm hungry 🍔 You challenged yourself to compile so much of what we've learned through the bootcamp into this project - celebrate that win! There are a few notes for considerations and comments that were out of place, but it was minimal. Reach out if clarifying anything would be helpful!

File: src/App.js (hunk 1 above)
Comment (gabrielwright1) [reply src/App.js:6]: There's so many useful methods!

File: src/App.js (hunk 2 above)
Comment (gabrielwright1) [reply src/App.js]: That's a good point, I'll definitely do that.

File: src/App.js (hunk 4 above)
Comment (gabrielwright1) [reply src/App.js]: Yep, that's a good idea for readability.

File: src/components/Card.js (hunk 7 above)
Comment (gabrielwright1) [reply src/components/Card.js:60]: 🙌🏾

File: src/styles/sass/partials/_typography.scss (hunk 8 above)
Comment (gabrielwright1) [reply src/styles/sass/partials/_typography.scss]: Haha yeah! Great catch!

File: src/styles/sass/partials/_typography.scss (hunk 9 above)
Comment (gabrielwright1) [reply src/styles/sass/partials/_typography.scss]: I borrowed my typography partial from my portfolio and completely forgot to update the comments. Thanks for catching this.

File: src/styles/sass/partials/_mixins.scss [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+	$line-height: normal,
+	$letter-spacing: normal
+) {
+	font-family: $font-family;
+	font-weight: $font-weight;
+	font-size: calc($font-size / 20) * 1rem;
+
+	@if $line-height != normal {
+		line-height: calc($line-height / 20) * 1rem;
+	} @else if $letter-spacing != normal {
```
Comment (gabrielwright1) [inline src/styles/sass/partials/_mixins.scss]: Split this "If" statement into two blocks, right now, only one of the two will pass.

**Comment by gabrielwright1** [comment]
Incorporated changes suggested by Christina and refactored the app into smaller components.

**Pull request merged**